var/spack/repos/builtin/packages/root/package.py

```diff
@@ -361,8 +364,8 @@ def cmake_args(self):
                 ['oracle', False],
                 ['pgsql', 'postgres'],
                 ['pythia6'],
-                ['pythia8', False],
-                ['python', self.spec.satisfies('+python')],
+                ['pythia8'],
+                ['python'],
```

Comment:
Is there a reason activation of Python support no longer depends on `+python` variant? This would not seem to be warranted for this change on its face, and if you don't also take account of the change here:
https://github.com/spack/spack/blob/0522c2d8403a3f589de78f2d8c4c4da4ade5af8c/var/spack/repos/builtin/packages/root/package.py#L423-L426
you could find ROOT building against native Python rather than the one specified in the concretized spec.

Reply:
I had the impression that an explicit `self.spec.satisfies('+python')` was not necessary for the same reason that e.g. `pythia6` doesn't need a `self.spec.satisifes('+pythia6')`, namely that [this complicated conditional in `_process_opts` will do the right thing](https://github.com/spack/spack/blob/0522c2d8403a3f589de78f2d8c4c4da4ade5af8c/var/spack/repos/builtin/packages/root/package.py#L494). Isn't that the case?

Reply:
You're right, of course: I should have looked at the full file more closely. I couldn't understand initially why the code was as it was in the first place, and then I saw this: https://github.com/spack/spack/commit/b98201593027850e23da6b65f9ba53474a4a2ed6#diff-09e58717bcd0dfb20a2714fdeaa39ec1L361-R364
_i.e._ the code originally had version clauses. As you say, the current state of L368 in in your PR is correct.